frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -145,3 +150,82 @@ export const useEventSourceList = (namespace: string) => {
   };
   return eventSourceList;
 };
+
+export const getEventSourceYaml = (namespace: string, eventSource: string): K8sResourceKind => {
+  let eventSourceModel: K8sKind;
+  const name = `${eventSource}`;
+  switch (eventSource) {
+    case 'camelsource':
+      eventSourceModel = EventSourceCamelModel;
+      break;
+    default:
+      eventSourceModel = {
+        apiGroup: KNATIVE_EVENT_SOURCE_APIGROUP_DEP,
+        apiVersion: 'v1alpha1',
+        kind: '',
+        label: '',
+        labelPlural: '',
+        plural: '',
+        abbr: '',
+      };
+  }
+  return {
+    apiVersion: `${eventSourceModel.apiGroup}/${eventSourceModel.apiVersion}`,
+    kind: eventSourceModel.kind,
+    metadata: {
+      name,
+      namespace,
+    },
+    spec: {
+      sink: {
```

Comment:
can we make use of `getEventSourcesDepResource` itself instead of this method

Reply:
updated the code for `getEventSourcesDepResource`. PTAL